plugin/plugin_test.go

```diff
@@ -56,117 +52,211 @@ func TestLogicOfRateLimit(t *testing.T) {
 	defer svr.Close()
 
 	requestURL := fmt.Sprintf("%v?id=1", svr.URL)
-	log.Println("requestURL", requestURL)
 
-	for i := 0; i < 1000; i++ {
-		if i%200 == 0 {
-			time.Sleep(time.Second * 1)
-		}
-		wg.Add(1)
-		go func(wg *sync.WaitGroup, mut *sync.Mutex, i int) {
-			defer wg.Done()
-			for j := 0; j < 1; j++ {
-				resp, err := svr.Client().Get(requestURL)
-				if err != nil {
-					fmt.Printf("Error: %s", err)
-					// t.Errorf("Error in %v, expected: %v, got: %v", test.url, test.expectedStatusCode, err.Error())
-				}
-				if resp.StatusCode == 200 {
-					mut.Lock()
-					results = append(results, fmt.Sprintf("PASS: i=%v, j=%v, time=%v", i, j, time.Since(initialTime).Milliseconds()))
-					mut.Unlock()
-				} else {
-					mut.Lock()
-					results = append(results, fmt.Sprintf("FAIL: i=%v, j=%v, time=%v", i, j, time.Since(initialTime).Milliseconds()))
-					mut.Unlock()
+	ticker := time.NewTicker(time.Second * 1)
+
+	j := 0
+
+	type Error struct {
+		Index int
+		Err   error
+	}
+
+	errChan := make(chan *Error, N)
+
+	for {
+		<-ticker.C
+
+		for i := 0; i < 200; i++ {
+
+			wg.Add(1)
+			go func(wg *sync.WaitGroup, i int) {
+				defer wg.Done()
+				for j := 0; j < 1; j++ {
+					resp, err := svr.Client().Get(requestURL)
+					errChan <- &Error{
+						Index: i,
+						Err:   err,
+					}
+					assert.Equal(t, 200, resp.StatusCode)
 				}
-			}
-		}(wg, mut, i)
+			}(wg, i)
+		}
+
+		j++
+
+		if j == 5 {
```

Comment:
Looks arbitrary, could you explain this?

Reply:
Sure!!
![image](https://github.com/VermaShivansh/coraza-ratelimit-plugin/assets/75882684/557aaeb5-e1bd-4717-90b5-986318240db4)
I have mentioned in the comments of the test case that this test case sends 200 requests every second for 5 times so it overalls to 1000reqs and they respond with 200.

Reply:
Thanks, I would move the 5 into a constant and reuse it below.

Reply:
sure, fixing it